### DISCUSSION TITLE
Request for early removal of disabled git-credential-manager formula

### DISCUSSION THREAD

**Discussion by ldennington** [Homebrew/discussions#4460] [Tap maintenance and Homebrew development]
### Description of issue

Hello!

My name is Lessley Dennington, and I am one of the maintainers of the [Git Credential Manager](https://github.com/git-ecosystem/git-credential-manager) (GCM) tool. We currently publish this tool as a [cask in an internally-maintained tap](https://github.com/microsoft/homebrew-git/blob/master/Casks/git-credential-manager-core.rb). However, we're making some updates to our release process and, as part of these changes, would like to start publishing GCM as a formula in `Homebrew/homebrew-core`. 

Unfortunately, we are currently unable to publish to `Homebrew/homebrew-core` because the name `git-credential-manager` is already claimed by an [existing formula](https://github.com/Homebrew/homebrew-core/blob/6524544aab85de5b5872793590bb0b8791794379/Formula/git-credential-manager.rb#L4). This formula is linked to the archived [Git Credential Manager for Mac and Linux](https://github.com/microsoft/Git-Credential-Manager-for-Mac-and-Linux) repository (which was also owned by my team) and, because of this, was [disabled about 8 months ago](https://github.com/Homebrew/homebrew-core/commit/e765b8134fabd93807f9f41d6e90a898e986dd1c).

I know formulae are automatically removed one year after being disabled. However, in light of our plans to begin publishing GCM as a formula with our next release (slated for Monday, May 8), I wanted to see whether it would be possible to remove this formula early to make the name `git-credential-manager` available to us. Please let me know if you have questions or need additional information, and thank you in advance for your help!

**Comment by SMillerDev** [comment]
I don't think we should remove a formula so it can be immediately succeeded by something else with the same name. I'd feel much more comfortable to add your software with a suffix now and add an alias for it later, making it clear that it's different software

**Reply by carlocab** [reply]
The new software with the same name is the successor of the software packages by the disabled formula, though. So reviving it with the new one seems reasonable.

**Reply by SMillerDev** [reply]
If it's the actual successor then we shouldn't remove the original either but just update the source.

**Reply by carlocab** [reply]
Yes, that's what I meant by "reviving it with the new one". The [archived repo](https://github.com/Microsoft/Git-Credential-Manager-for-Mac-and-Linux) README says

> This project has been superceded by [Git Credential Manager Core](https://github.com/microsoft/Git-Credential-Manager-Core).

**Reply by gromgit** [reply]
@ldennington FWIW, I tried doing exactly that with https://github.com/Homebrew/homebrew-core/pull/121648. It built fine locally, but I couldn't figure out the NuGet errors in CI, and eventually gave up. That was two releases back (2.0.886) , so maybe you can use my previous effort as a starting point.

**Reply by ldennington** [reply]
Thanks all for the helpful responses. I was not clear enough in my original message that:

1. The Git Credential Manager tool _is_ the official successor to the Git Credential Manager for Mac and Linux tool currently built by the formula.
2. My team owns both the current GCM tool and the archived tool.

Based on these responses, I will plan to update the existing formula, building off the excellent start provided by @gromgit.
